src/main/java/sirius/pasta/noodle/compiler/CompilationContext.java

```diff
@@ -0,0 +1,443 @@
+     * <p>
+     * Next to the classic <tt>Class.forName</tt> this also supports aliases like <tt>String</tt> for
+     * <tt>java.lang.String</tt>.
+     *
+     * @param position the position for error reporting
+     * @param typeName the type name to resolve
+     * @return a Java class for the given type name
+     */
+    public Class<?> resolveClass(Position position, String typeName) {
+        Class<?> result = tryResolveClass(typeName).orElse(null);
+        if (result == null) {
+            error(position, "Cannot resolve '%s' to a Java class", typeName);
+            result = void.class;
+        }
+        return result;
+    }
+
+    /**
+     * Tries to resolve a string (name) into a Java class.
+     * <p>
+     * Next to the classic <tt>Class.forName</tt> this also supports aliases like <tt>String</tt> for
+     * <tt>java.lang.String</tt>.
+     *
+     * @param typeName the type name to resolve
+     * @return a Java class for the given type name wrapped as optional or an empty optional, if no matching class was
+     * found.
+     */
+    public Optional<Class<?>> tryResolveClass(String typeName) {
+        Class<?> result = importedClasses.get(typeName);
+        if (result != null) {
+            return Optional.of(result);
+        }
+
+        result = getClassAliases().get(typeName);
+        if (result != null) {
+            return Optional.of(result);
+        }
+
+        try {
+            return Optional.of(Class.forName(typeName));
+        } catch (ClassNotFoundException e) {
+            Exceptions.ignore(e);
+            return Optional.empty();
+        }
+    }
+
+    private static Map<String, Class<?>> getClassAliases() {
+        if (aliases == null) {
+            Map<String, Class<?>> aliasMap = new HashMap<>();
+            aliasProviders.forEach(p -> p.collectAliases(aliasMap::put));
+            aliases = aliasMap;
+        }
+
+        return Collections.unmodifiableMap(aliases);
+    }
+
+    /**
+     * Determines if the type <tt>to</tt> is assignable from the given object.
+     * <p>
+     * In contrast to {@link Class#isAssignableFrom(Class)}, this also handles autoboxing appropriately.
+     *
+     * @param from the object to assign
+     * @param to   the type to assign to
+     * @return <tt>true</tt> if the object is assignable, <tt>false</tt> otherwise
+     */
+    public static boolean isAssignable(Object from, Class<?> to) {
+        if (from == null) {
+            return !to.isPrimitive();
+        }
+
+        return isAssignableTo(from.getClass(), to);
+    }
+
+    /**
+     * Unboxes the given class.
+     * <p>
+     * This will return the boxed type for a primitive class like <tt>Integer</tt> for <tt>int</tt>.
+     *
+     * @param type the classto unbox
+     * @return the unboxed version of the given class or the class itself if it wasn't a primitive
+     */
+    public static Class<?> unboxClass(Class<?> type) {
```

Comment:
Isnt this actually autoboxing instead of unboxing?

autoboxing = primitive -> object wrapper
unboxing = object wrapper -> primitive